### PR TITLE
Introduce AllowAllMiddleware for dummy authorization

### PR DESCRIPTION
**Pull request by Xenopathic**
Useful for testing, such as the codegen sample tests which define authorization but always need to succeed.

Since the auth stuff was getting quite large, I've also moved it to a separate file and have pulled in Authorization and AuthData as `pub use` for compatibility.

### CODE REVIEW

**Comment by mirw** [comment]
Do we need to bump the version number in Cargo.toml?

File: src/auth.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+/// Dummy implementation of an Iron middleware to insert authorization data,
+/// allowing all access to an endpoint with the subject "alice".
+#[derive(Debug)]
+pub struct AllowAllMiddleware;
+
+impl iron::middleware::BeforeMiddleware for AllowAllMiddleware {
+    fn before(&self, req: &mut iron::Request) -> iron::IronResult<()> {
+        req.extensions.insert::<Authorization>(Authorization {
+            subject: "alice".to_string(),
```
Comment (mirw) [inline src/auth.rs]: Should we make this a parameter on the constructor?

**Comment by mirw** [comment]
A couple of comments.  Otherwise looks fine.

**Comment by Xenopathic** [comment]
@mirw All comments resolved, including the point about using `Option<_>` for the scopes

**Comment by mirw** [comment]
Looks good.

**Pull request merged**